### PR TITLE
Fixes #2888 : First time ever tapping "Nearby": Tab does not show

### LINKED ISSUE #2888: First time ever tapping "Nearby": Tab does not show
**Steps to reproduce:** 

1. Install app on new phone
2. Log in
3. Tap "nearby"
4. Location permission popup appear, accept
5. Nearby tab is not opened (nearest location bar is shown, though)

Workaround: Tap on "Nearby" again.

**Commons app version:** 
2.10.2-debug-master~84637c5fe

**Screen-shots:** 
Screenshot from 2019-04-17 17-38-54

### PR DESCRIPTION
**Pull request by ahmedmamdouh13**
Steps to reproduce:

    Install app on new phone
    Log in
    Tap "nearby"
    Location permission popup appear, accept
    Nearby tab is not opened (nearest location bar is shown, though)

Workaround: Tap on "Nearby" again.

Fixes #2888 : First time ever tapping "Nearby": Tab does not show

took requesting location permission off the main thread to stop delaying UI elements 

Tested on Huawei P9 lite API level 24 | Nexus 5 with API level  25.

untitled

### CODE REVIEW

**Comment by ahmedmamdouh13** [comment]
because opening the tab while requesting permission is a lot on the main thread to handle, especially when the app is freshly installed requesting permission for the first time from the OS takes longer than usual , doing so on the main thread causes this lag which is viewpager not responding to Tab clicking.

**Comment by maskaravivek** [comment]
I don't think threading the issue in this case. Check out `NearbyFragment`. My guess is that the fragment is returning early if the location is not present.

**Comment by ahmedmamdouh13** [comment]
okay but let me tell you what i tried , i tried to do not request permission at all , what happened was that everything went smoothly and Nearby tab opened with no bug , another thing is that even when requesting the permission on another thread their was an obvious delay of 1 second or less which explains why it was causing the bug.

File: app/src/main/java/fr/free/nrw/commons/upload/UploadModel.java [hunk 1]
```diff
@@ -209,6 +209,7 @@ public void next() {
     void setCurrentTitleAndDescriptions(Title title, List<Description> descriptions) {
         setCurrentUploadTitle(title);
         setCurrentUploadDescriptions(descriptions);
+        store.putString("Title", title.toString());
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadModel.java]: Is this change related?

File: app/src/main/java/fr/free/nrw/commons/nearby/NearbyFragment.java [hunk 2]
```diff
@@ -602,7 +604,13 @@ private void registerLocationUpdates() {
      */
     private void requestLocationPermissions() {
         if (!getActivity().isFinishing()) {
-            locationManager.requestPermissions(getActivity());
+            new Single<Object>(){
```
Comment (maskaravivek) [inline app/src/main/java/fr/free/nrw/commons/nearby/NearbyFragment.java]: Dispose of the single

File: app/src/main/java/fr/free/nrw/commons/upload/UploadModel.java (hunk 1 above)
Comment (ahmedmamdouh13) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadModel.java]: no , this change of another issue , i didn't know that every pull request needs a new branch , i'll try to fix it.

File: app/src/main/java/fr/free/nrw/commons/nearby/NearbyFragment.java (hunk 2 above)
Comment (ahmedmamdouh13) [reply app/src/main/java/fr/free/nrw/commons/nearby/NearbyFragment.java]: okay got it.

**Comment by maskaravivek** [comment]
@ahmedmamdouh13 I discussed with @neslihanturan and @ashishkumar468 and we feel that the root cause of the issue is that either the lifecycle events are not handled properly or some method is returning early. 

Can I take up this issue and spend a couple of days trying to figure out the root cause? I suggest that you take up another beginner-friendly issue for the time being. :)

**Comment by ahmedmamdouh13** [comment]
@maskaravivek yes of course , i know it was not an ideal solution but it's their if the problem persists , thank you for informing me 🙏☺️

**Comment by neslihanturan** [comment]
Hi @maskaravivek , seems like this issue is solved by a technically better approach at #2925 . Please join the discussion there if you want to. Thanks for your contribution.

**Review by akkizle — APPROVED** [review]
What isvthis
